### DISCUSSION TITLE
Apple Provider: Unable to Retrieve User Name on Initial Login (Name: Undefined)

### DISCUSSION THREAD

**Discussion by AyrDS** [nextauthjs/next-auth#12231] [Help]
I'm encountering an issue with the Apple OAuth provider where I cannot retrieve the user's name on the first login. When a user signs in with Apple for the first time, I expect to receive the user's name, but it consistently arrives as `name: undefined` in my application.

I've configured my authentication settings in a shared `auth.config.ts` file, and I believe the setup is correct. However, I'm still unable to get the user's name, even though this should be available on the first login.

Could anyone help me understand what might be going wrong or if there is a configuration or additional step needed to ensure the name field is populated?

Here is a snippet from my `auth.config.ts` file where I set up the Apple provider.
```
AppleProvider({
      clientSecret: env.APPLE_CLIENT_SECRET,
      allowDangerousEmailAccountLinking: true,
      clientId: env.APPLE_CLIENT_ID,
      wellKnown: 'https://appleid.apple.com/.well-known/openid-configuration',
      checks: ['pkce'],
      token: {
        url: 'https://appleid.apple.com/auth/token',
      },
      authorization: {
        url: 'https://appleid.apple.com/auth/authorize',
        params: {
          scope: 'email name',
          response_type: 'code',
          response_mode: 'form_post',
          state: crypto.randomUUID(),
        },
      },
      client: {
        token_endpoint_auth_method: 'client_secret_post',
      },
    })
   ```
```
cookies: {
    sessionToken: {
      name: 'booking-user-session',
      options: {
        httpOnly: true,
        sameSite: 'lax',
        path: '/',
        domain: env.NEXT_PUBLIC_ENVIRONMENT === 'production' ? '.example.com' : undefined, // Solo aplica en producción
        secure: env.NEXT_PUBLIC_ENVIRONMENT === 'production', // Solo aplica secure en producción
      },
    },
    pkceCodeVerifier: {
      name: 'next-auth.pkce.code_verifier',
      options: {
        httpOnly: true,
        sameSite: 'none',
        path: '/',
        domain: env.NEXT_PUBLIC_ENVIRONMENT === 'production' ? '.example.com' : undefined, // Solo aplica en producción
        secure: true,
      },
    },
  }
```

**Comment by danielpl10** [comment]
@AyrDS Hello, did you have any success adding Apple Provider to your project, Please if is yes, can you share your final auth config file and any other changes needed? Thanks
